proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,797 @@
+# Machine readable JSON output for dotnet list package
+
+* Status: **In Review**
+* Author: [Erick Yondon](https://github.com/erdembayar)
+* GitHub Issue [7752](https://github.com/NuGet/Home/issues/7752)
+
+## Problem background
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Many organization are required by [regulation](https://www.whitehouse.gov/briefing-room/presidential-actions/2021/05/12/executive-order-on-improving-the-nations-cybersecurity/) to audit packages that they're using in repository.
+Currently there's no easy way produce Software Bill of Material (SBOM) output which can be consumed by another auditing system or historic keeping.
+
+* Parse-friendly output. Other PMP like Npm already have it(`npm ls --parseable` and `npm ls --json`).
+* Useful for CI/CD auditing(compliance, security ..)
+  * Produce Software Bill of Material (SBOM) (compliance, historic keeping)
+  * Enhancing Software Supply Chain Security
+    * check any vulnerable packages
+    * check any deprecated packages
+    * Check any outdated packages
+  * Check license compliance
+  * Resolve dependency issue (detect duplicate packages, detect new dependency introduced or existing one removed etc ..)
+  * Making the output machine-readable unlocks additional tooling and automation scenarios in CI/CD pipeline above scenarios.
+
+## Who are the customers
+
+Anyone (government/private enterprises, security experts, individual contributors) who wants to consume `dotnet list package` output for auditing tool or historic keeping, CI/CD orchestrating.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+### --json option
+
+Ability to use new `--json` option for all `dotnet list package` commands to ensure JSON-formatted output is emitted to the console.
+
+```dotnetcli
+dotnet list [<PROJECT>|<SOLUTION>] package [--config <SOURCE>]
```

Comment:
`dotnet list` shows _project_ references and package references.  Are we going to make both types display as JSON?  If not, we'll have to figure out way to make the `--format json` only show when a users sees the usage for `dotnet list <PROJECT> package`

Reply:
It only applies to `package` only. Considering below works I believe it's possible.

```
dotnet list .\MyProjectA\MyProjectA.csproj package
Project 'MyProjectA' has the following package references
   [netcoreapp3.1]:
   Top-level Package                      Requested             Resolved
   > Microsoft.Extensions.Primitives      [1.0.0, 5.0.0]        1.0.0
   > NuGet.Commands                       4.8.0-preview3.5278   4.8.0-preview3.5278
   > Text2Xml.Lib                         [1.1.2, 2.0.0)        1.1.2

```

Reply:
`dotnet list <PROJECT> package` doesn't display any project reference, only package references. See above example.